CoreScriptsRoot/ServerCoreScripts/ClientChat/ChatBar.lua

```diff
@@ -68,8 +86,79 @@ function methods:CreateGuiObjects(targetParent)
 	TextLabel.TextStrokeTransparency = 1
 	TextLabel.TextTransparency = 0.4
 
-	TextBox.Focused:connect(function() TextLabel.Visible = false end)
-	TextBox.FocusLost:connect(function() TextLabel.Visible = (TextBox.Text == "") end)
+	TextBox.TextColor3 = TextLabel.TextColor3
+	TextBox.TextStrokeTransparency = TextLabel.TextStrokeTransparency
+	TextBox.TextTransparency = TextLabel.TextTransparency
+
+	--TextBox.TextColor3 = Color3.fromRGB(112 * 0.7, 110 * 0.7, 106 * 0.7) --Color3.fromRGB(112, 110, 106)
+
+	local function UpdateOnFocusStatusChanged(isFocused)
+		if (isFocused) then
+			TextLabel.Visible = false
+			MessageModeTextBox.Visible = true
+		else
+			local setVis = (TextBox.Text == "")
+			TextLabel.Visible = setVis
+			MessageModeTextBox.Visible = not setVis
+		end
+	end
+
+	TextBox.Focused:connect(function() UpdateOnFocusStatusChanged(true) end)
+	TextBox.FocusLost:connect(function() UpdateOnFocusStatusChanged(false) end)
+
+	UserInputService.InputBegan:connect(function(inputObj, gpe)
```

Comment:
comment what this does. if in channel, empty input textbox, go back to "allchat".
should this check what the current channel is (to avoid unnecessary network traffic)